src/codeu/chat/server/Model.java

```diff
@@ -67,6 +69,27 @@ public int compare(Time a, Time b) {
   private final Store<Time, Message> messageByTime = new Store<>(TIME_COMPARE);
   private final Store<String, Message> messageByText = new Store<>(STRING_COMPARE);
 
+  private final Store<User, HashSet<Uuid>> userInterestByUser = new Store<>(null);
+  private final Store<User, HashSet<Uuid>> convoInterestByUser = new Store<>(null);
+
+  public void addUserInterest(User user, User interested, UserInterests userInterests) {
+    userInterests.interestedInUsers.add(interested.id);
+    userInterestByUser.insert(user, userInterests.interestedInUsers);
+  }
+
+  public void addConversationInterest(User user, ConversationHeader interested, UserInterests userInterests) {
+    userInterests.interestedInConvos.add(interested.id);
+    convoInterestByUser.insert(user, userInterests.interestedInConvos);
+  }
```

Comment:
This function implementation should be very similar, but the last line will be something like:

`userInterest.interestedInConvos.add(interested.id);`